examples/usvg_viewer/src/main.rs

```diff
@@ -0,0 +1,212 @@
+    use winit::{event::*, event_loop::ControlFlow};
+    let mut render_cx = RenderContext::new().unwrap();
+    let size = window.inner_size();
+    let mut surface = render_cx
+        .create_surface(&window, size.width, size.height)
+        .await;
+    let device_handle = &render_cx.devices[surface.dev_id];
+    let mut renderer = Renderer::new(&device_handle.device).unwrap();
+    let mut current_frame = 0usize;
+    let mut scene = Scene::new();
+    let mut cached_svg_scene = vec![];
+    cached_svg_scene.resize_with(svg_files.len(), || None);
+    let mut transform = Affine::IDENTITY;
+    let mut mouse_down = false;
+    let mut prior_position = Vec2::default();
+    let mut last_title_update = Instant::now();
+    // We allow looping left and right through the svgs, so use a signed index
+    let mut svg_ix: i32 = 0;
+    // These are set after choosing the svg, as they overwrite the defaults specified there
+    event_loop.run(move |event, _, control_flow| match event {
+        Event::WindowEvent {
+            ref event,
+            window_id,
+        } if window_id == window.id() => match event {
+            WindowEvent::CloseRequested => *control_flow = ControlFlow::Exit,
+            WindowEvent::KeyboardInput { input, .. } => {
+                if input.state == ElementState::Pressed {
+                    match input.virtual_keycode {
+                        Some(VirtualKeyCode::Left) => {
+                            svg_ix = svg_ix.saturating_sub(1);
+                            transform = Affine::IDENTITY
+                        }
+                        Some(VirtualKeyCode::Right) => {
+                            svg_ix = svg_ix.saturating_add(1);
+                            transform = Affine::IDENTITY
+                        }
+                        Some(VirtualKeyCode::Space) => transform = Affine::IDENTITY,
+                        Some(VirtualKeyCode::Escape) => {
+                            *control_flow = ControlFlow::Exit;
+                        }
+                        _ => {}
+                    }
+                }
+            }
+            WindowEvent::Resized(size) => {
+                render_cx.resize_surface(&mut surface, size.width, size.height);
+                window.request_redraw();
+            }
+            WindowEvent::MouseInput { state, button, .. } => {
+                if button == &MouseButton::Left {
+                    mouse_down = state == &ElementState::Pressed;
+                }
+            }
+            WindowEvent::MouseWheel { delta, .. } => {
+                let modifier = if let MouseScrollDelta::PixelDelta(delta) = delta {
+                    delta.y * 0.001
+                } else if let MouseScrollDelta::LineDelta(_, y) = delta {
+                    *y as f64 * 0.1
+                } else {
+                    0.0
+                };
+                transform = Affine::translate(prior_position)
+                    * Affine::scale(1.0 + modifier)
```

Comment:
Don't know how to address your concerns. Open to suggestions.

Reply:
Just passing by, this is very intuitive to me so I was surprised it was not for others. Trying my best to be pedagogical.

You can think of scrolling by adding delta times increment (multiplication) as repeatedly scrolling by the increment, delta times (repeated addition).
The equivalent for scaling would be doing delta times the scale.

But scaling is a multiplicative operation (scrolling is additive).
i.e:
- for scrolling : not scrolling is scrolling by 0 (like addition), and the inverse, scrolling back is adding the negation (like addition).
- for scaling: not scaling is scaling by 1 (like multiplication), and the inverse, scaling back is multiplying by the reciprocal (like multiplication).
 
So the repeated addition of an increment becomes a repeated multiplication by a base.

Which is also known as exponentiation: scale(pow(base, times)). [times is delta]
e.g pow(base,0) = 1, pow(base,-times) = 1/pow(base,times) so it has the properties we want
You want your base to be >1 so that pow(base,times) is >1 for positive times.
And close to 1 (<2) to stay sane.

Reply:
Mmm, right. Now scrolling 10 "lines" will scale the image by `1.05^(-10) = 0.61` instead of `1-1 = 0`.

Reply:
@DJMcNab Did the update address your concerns?

Reply:
Yeah, I think so.